cmd/antrea-controller/controller.go

```diff
@@ -100,23 +103,50 @@ func run(o *Options) error {
 
 	go apiServer.GenericAPIServer.PrepareRun().Run(stopCh)
 
+	if o.config.EnablePrometheusMetrics {
+		go initializePrometheusMetrics()
```

Comment:
Does it really need to be executed in another go routine? It's not a long running call, right?

Reply:
Indeed. That was needed when a dedicated listener was initialized within initializePrometheusMetrics()